data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessor.java

```diff
@@ -136,66 +146,124 @@ private void parseFile(String filePath){
 
     @Override
     public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
-        //todo
-        for(final Record<Event> record : records) {
+        for (final Record<Event> record : records) {
             final Event recordEvent = record.getData();
-            if (Objects.nonNull(translateProcessorConfig.getMapWhen()) && !expressionEvaluator.evaluateConditional(translateProcessorConfig.getMapWhen(), recordEvent)) {
+            if (Objects.nonNull(translateProcessorConfig.getTranslateWhen()) && !expressionEvaluator.evaluateConditional(translateProcessorConfig.getTranslateWhen(), recordEvent)) {
                 continue;
             }
             try {
-                String matchKey = record.getData().get(translateProcessorConfig.getSource(), String.class);
-                if(matchesIndividualEntry(record, matchKey) || matchesRangeEntry(record, matchKey) || matchesPatternEntry(record, matchKey)){
-                    continue;
+                if (Objects.nonNull(translateProcessorConfig.getIterateOn())) {
+                    List<Map<String, Object>> nestedObjects = recordEvent.get(translateProcessorConfig.getIterateOn(), List.class);
```

Comment:
I think we should also support when the List is a `List<String>` . To do so we could require the `source` and the `iterate_on` value to be the same, or we could detect from `source` by trying to convert the value to `List<String>`

Reply:
As per the design I have moved that functionality to source option. Source can take a single string source or List of sources. The idea is to have each option have its own functionality that sources are sources are iterate_on is a array of objects that needs the mappings. We can discuss it further if you think we need a design change.